stakingModel/test/GoodReserve.e2e.test.js

```diff
@@ -0,0 +1,193 @@
+    // Verifies that the ExecuteProposal event has been emitted
+    expect(executeProposalEventExists).to.be.true;
+    await setBlockInterval.setBlockInterval();
+    const blockInterval = await goodReserve.blockInterval();
+    expect(blockInterval.toString()).to.be.equal('5759');
+  });
+
+  it("should be able to set the contribution address", async () => {
+    const executeProposalEventExists = await proposeAndRegister(setContributionAddress.address,
+      registrar, proposalId, absoluteVote, avatarAddress, founder);
+    // Verifies that the ExecuteProposal event has been emitted
+    expect(executeProposalEventExists).to.be.true;
+    await setContributionAddress.setContributionAddress();
+    const contributionAddressT = await goodReserve.contribution();
+    expect(contributionAddressT).to.be.equal(contribution.address);
+  });
+
+  it("should returned fixed 0.0001 market price", async () => {
+    const gdPrice = await goodReserve.currentPrice(cDAI.address);
+    const cdaiWorthInGD = gdPrice.mul(new BN("100000000", 10));
+    const gdFloatPrice = gdPrice.toNumber() / 10 ** 8; //cdai 8 decimals
+    expect(gdFloatPrice).to.be.equal(0.0001);
+    expect(cdaiWorthInGD.toString()).to.be.equal("1000000000000"); //in 8 decimals precision
+    expect(cdaiWorthInGD.toNumber() / 10 ** 8).to.be.equal(10000);
+  });
+
+  it("should be able to set the reserve as a minter", async () => {
+    const executeProposalEventExists = await proposeAndRegister(addMinter.address,
+    registrar, proposalId, absoluteVote, avatarAddress, founder);
+    // Verifies that the ExecuteProposal event has been emitted
+    expect(executeProposalEventExists).to.be.true;
+    await addMinter.addMinter();
+    const isMinter = await goodDollar.isMinter(goodReserve.address);
+    expect(isMinter).to.be.true;
+  });
+
+  it("should be able to buy gd with cDAI and reserve should be correct", async () => {
+    let amount = 1e8;
+    await dai.mint(web3.utils.toWei("100", "ether"));
+    dai.approve(cDAI.address, web3.utils.toWei("100", "ether"));
+    await cDAI.mint(web3.utils.toWei("100", "ether"));
+    let reserveToken = await marketMaker.reserveTokens(cDAI.address);
+    let reserveBalanceBefore = reserveToken.reserveSupply;
+    const cDAIBalanceReserveBefore = await cDAI.balanceOf(goodReserve.address);
+    await cDAI.approve(goodReserve.address, amount);
+    await goodReserve.buy(cDAI.address, amount, 0);
+    reserveToken = await marketMaker.reserveTokens(cDAI.address);
+    let reserveBalanceAfter = reserveToken.reserveSupply;
+    const cDAIBalanceReserveAfter = await cDAI.balanceOf(goodReserve.address);
+    expect((cDAIBalanceReserveAfter - cDAIBalanceReserveBefore).toString()).to.be.equal(
+      amount.toString()
+    );
+    expect((reserveBalanceAfter - reserveBalanceBefore).toString()).to.be.equal(
+      amount.toString()
+    );
+  });
+
+  it("should be able to buy gd with cDAI and the price should be the same", async () => {
```

Comment:
WHAT? how can price be the same after selling/buying?

Reply:
it was a test for the precision and not for the price.
changed at db43a7633912fed7aa89a7ff580fbe5238d650b5